agent/coordinate_endpoint_test.go

```diff
@@ -85,9 +85,14 @@ func TestCoordinate_Nodes(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 
-	coordinates := obj.(structs.Coordinates)
-	if coordinates == nil || len(coordinates) != 0 {
-		t.Fatalf("bad: %v", coordinates)
+	// Check that coordinates are empty before registering a node
+	coordinates, ok := obj.(structs.Coordinates)
+	if !ok {
+		t.Fatalf("expected: structs.Coordinates, received: %v", coordinates)
```

Comment:
The expected value here would just show up as a string so it's not super useful to say expected `structs.Coordinates`. Maybe something a little more specific like `structs.Coordinates should not be nil`.

Reply:
@alvin-huang Good point about printing `coordinates` as a string, we should avoid that.

I think we should also guard against `obj` being of the wrong type here since it's coming back as an empty interface. 

Maybe this should print `obj` instead of coordinates if `!ok`:
```suggestion
		t.Fatalf("expected: structs.Coordinates, received: %v", obj)
```

That way if `obj` is nil it prints `expected: structs.Coordinates, received: <nil>` or if it's of the wrong type it'll show the contents.

Reply:
Here's an example for reference: https://play.golang.org/p/O7MDEFrXf6h

Reply:
This is a really good idea! Thank you!